app/reflexes/admin/orders_reflex.rb

```diff
@@ -111,14 +115,17 @@ def set_param_for_controller
     end
 
     def all_distributors_can_invoice?(orders)
-      distributors = orders.map(&:distributor).uniq.reject(&:can_invoice?)
+      distributor_ids = orders.map(&:distributor_id)
+      Enterprise.where(id: distributor_ids, abn: nil).empty?
```

Comment:
Much better! In theory, that could still be a lot of ids. We could use a sub-query instead but I'm happy with this solution.